osu.Game/Rulesets/RulesetInfo.cs

```diff
@@ -49,6 +49,20 @@ public bool Equals(RulesetInfo? other)
 
         public bool Equals(IRulesetInfo? other) => other is RulesetInfo b && Equals(b);
 
+        public int CompareTo(RulesetInfo other)
+        {
+            // Official rulesets are always given precedence for the time being.
+            if (OnlineID >= 0)
+            {
+                if (other.OnlineID >= 0)
+                    return OnlineID.CompareTo(other.OnlineID);
+
+                return -1;
+            }
+
```

Comment:
I think this needs an `if (other.OnlineID >= 0) return 1` condition to be invertible. Or this could be rewritten into:
```
if (OnlineID >= 0 && other.OnlineID >= 0)
    return OnlineID.CompareTo(other.OnlineID);

// Official rulesets are always given precedence for the time being.
if (OnlineID >= 0)
    return -1;
if (other.OnlineID >= 0)
    return 1;

return string.Compare(ShortName, other.ShortName, StringComparison.Ordinal);
```

Reply:
Have updated to that implementation. Not sure how to test the potential fail case there, if there was one.